### PR TITLE
Time Accounting by Band

### PR DESCRIPTION
**Pull request by swalker2m**
Changes the `timeCharge` field at the `Program` level to a `BandedTime`, which groups a `ScienceBand` and a `CategorizedTime`.  In other words, it groups the time charges by bands at the program level.

__This will presumably break Explore.__

### CODE REVIEW

File: modules/schema/src/main/scala/lucuma/odb/data/BandedTime.scala [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+case class BandedTime(
+  band: Option[ScienceBand],
+  time: CategorizedTime
+)
+
+object BandedTime:
+  val Empty: BandedTime = BandedTime(None, CategorizedTime.Zero)
+
+  given Eq[BandedTime] =
```
Comment (tpolecat) [inline modules/schema/src/main/scala/lucuma/odb/data/BandedTime.scala:18]: Might be useful to define `Order` here. Looks like it's defined for the member types.

**Review by tpolecat — APPROVED** [review]
Looks good. 👍🏻

**Comment by toddburnside** [comment]
> This will presumably break Explore.

As far as I can see, Explore doesn't look at the `timeCharge`

File: modules/schema/src/main/scala/lucuma/odb/data/BandedTime.scala (hunk 1 above)
Comment (swalker2m) [reply modules/schema/src/main/scala/lucuma/odb/data/BandedTime.scala]: :+1: Thanks, that's a good point.  I've updated it but will add property tests when I move this to core.

**Pull request merged**